File: app/src/main/java/net/mmho/photomap2/PhotoListAdapter.java
package net.mmho.photomap2;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;

import java.util.ArrayList;
import java.util.Locale;

import rx.Observable;
import rx.Subscription;
import rx.android.schedulers.AndroidSchedulers;
import rx.schedulers.Schedulers;

class PhotoListAdapter extends ArrayAdapter<PhotoGroup> {

    private final int resource;
    private final LayoutInflater inflater;

    public PhotoListAdapter(Context context, int resource, ArrayList<PhotoGroup> objects) {
        super(context, resource, objects);
        this.resource = resource;
        inflater = (LayoutInflater)context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        View v;
        if(convertView!=null){
            v = convertView;
        }
        else {
            v = inflater.inflate(resource,null);
        }
        if(position < getCount()) {
            PhotoGroup g = getItem(position);
            ((PhotoCardLayout) v).setData(g);
        }
        return v;
    }
}

File: app/src/main/kotlin/net/mmho/photomap2/PhotoListAdapter.kt
package net.mmho.photomap2

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter

import java.util.ArrayList

internal class PhotoListAdapter(context: Context, private val resource: Int, objects: ArrayList<PhotoGroup>)
        : ArrayAdapter<PhotoGroup>(context, resource, objects) {
    private val inflater: LayoutInflater

    init {
        inflater = context.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
    }

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        val v: View =
            if(position!=0 && convertView!=null)convertView else inflater.inflate(resource, null)
        if (position < count) (v as PhotoCardLayout).setData(getItem(position))
        return v
    }
}
